pkg/controller/worker/machines.go

```diff
@@ -145,10 +145,16 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				"region":           w.worker.Spec.Region,
 				"availabilityZone": zone,
 				"machineType":      pool.MachineType,
-				"keyName":          infrastructureStatus.Node.KeyName,
-				"networkID":        infrastructureStatus.Networks.ID,
-				"podNetworkCidr":   extensionscontroller.GetPodNetwork(w.cluster),
-				"securityGroups":   []string{nodesSecurityGroup.Name},
+				"nodeTemplate": machinev1alpha1.NodeTemplate{
+					Capacity:     pool.NodeTemplate.Capacity,
```

Comment:
Is it guaranteed that `pool.NodeTemplate` is always set mean `!= nil`?

Reply:
yes , although its an optional field in the Worker API for backward compatibility , but its always set on the worker resource by gardenlet [here] (https://github.com/gardener/gardener/blob/153ebe7320e112211f05df97eabb71e34bfe8e62/pkg/operation/botanist/component/extensions/worker/worker.go#L231)

Reply:
Thanks for explaining.

Reply:
I think this field was added 8days ago and will be available in Gardener v1.37.0, right? But you should also consider cases where latest provider-openstack extensios runs against Gardener < v1.37.0. Hence, I guess you need proper handling when this field is not set (nil).

Reply:
Yes @ialidzhikov , its a valid point. I'll add this field only when its not `nil` now. Thanks